src/shared/hooks/useMovies.tsx

```diff
@@ -1,20 +1,14 @@
-import { useState, useEffect } from 'react';
+import { useQuery } from '@tanstack/react-query';
 
+import movieService, { GetMoviesParams } from '../../services/movieService';
 import { Movie } from '../../interfaces/movie.interface';
-import movieService from '../../services/movieService';
 
-export const useMovies = () => {
-  const [movies, setMovies] = useState<Movie[]>([]);
+export const useMovies = (params?: GetMoviesParams) => {
+  const moviesQuery = useQuery<Movie[], Error>({
+    queryKey: ['movies', params],
+    queryFn: () => movieService.getMovies(params),
+  });
+  const { data: movies, isLoading, error } = moviesQuery;
 
-  useEffect(() => {
-    movieService.getMovies()
-      .then((data) => {
-        setMovies(data);
-      })
-      .catch(error => {
-        console.error(error);
-      })
-  }, []);
-
-  return { movies };
-}
+  return { movies, isLoading, error };
+};
```

Comment:
**nitpick:** It's better to return `useQuery` from this hook

Reply:
done